CoreUtilities/src/au/gov/asd/tac/constellation/utilities/visual/VisualProcessor.java

```diff
@@ -304,7 +304,17 @@ private void processChangeSet(final Collection<VisualChange> changes, final Visu
             if (!masterChangeMap.containsKey(getMasterProperty(change.property))) {
                 masterChangeMap.put(getMasterProperty(change.property), new ArrayList<>());
             }
-            masterChangeMap.get(getMasterProperty(change.property)).add(change);
+            final List<VisualChange> currentChanges = masterChangeMap.get(getMasterProperty(change.property));
+            boolean equivilantChangePresent = false;
+            for (final VisualChange currentChange : currentChanges) {
+                if (currentChange.hasSameChangeList(change)){
+                    equivilantChangePresent = true;
+                    break;
+                }
+            }
+            if (!equivilantChangePresent) {
+                currentChanges.add(change);
+            }
```

Comment:
@Nova-2119 many thanks for the continued effort to help improve performance. However with this change I don't really get how its working. So `currentChanges` is a curated list of unique changes but this method still returns the unmodified `masterChangeMap` so I don't see how this is going to help. I will do some more debugging to see if I've missed something but just looking at the code this doesn't seem like its going to help.

Reply:
Its working so I take it back. I would still like to understand how this change works though.

FYI to debug I added the following just before the return at the end:
```
        for (final Map.Entry<VisualProperty, List<VisualChange>> entry : masterChangeMap.entrySet()) {
            if (!"EXTERNAL_CHANGE".equals(entry.getKey().name())) {
                LOG.log(Level.INFO, "{0} -> {1}", new Object[]{entry.getKey().name(), entry.getValue().size()});
            }
        }        
```

So `master` shows the following when deleting some nodes:

```
FINE [au.gov.asd.tac.constellation.plugins.PluginExecution]: Delete Selection
INFO [au.gov.asd.tac.constellation.utilities.visual.VisualProcessor]: VERTICES_REBUILD -> 2
INFO [au.gov.asd.tac.constellation.utilities.visual.VisualProcessor]: CONNECTIONS_REBUILD -> 2
```

And this branch reduces it to just 1 so it looks like:

```
FINE [au.gov.asd.tac.constellation.plugins.PluginExecution]: Delete Selection
INFO [au.gov.asd.tac.constellation.utilities.visual.VisualProcessor]: VERTICES_REBUILD -> 1
INFO [au.gov.asd.tac.constellation.utilities.visual.VisualProcessor]: CONNECTIONS_REBUILD -> 1
```